docs/developers/ESLint_Plugins.mdx

```diff
@@ -0,0 +1,91 @@
+
+:::tip
+See [**`eslint-plugin-example-typed-linting`**](https://github.com/typescript-eslint/examples/tree/39a6f7bb310645c88382a25aa8bed719ccd61a0c/packages/eslint-plugin-example-typed-linting) for an example plugin that supports typed linting.
+:::
+
+## Package Dependencies
+
+Your plugin should have the following `package.json` entries.
+
+For all `@typescript-eslint` and `typescript-eslint` packages, keep them at the same semver versions.
+As an example, you might set each of them to `^8.1.2` or `^7.12.0 || ^8.0.0`.
+
+### `dependencies`
+
+[`@typescript-eslint/utils`](../packages/Utils.mdx) is required for the [`RuleCreator` factory to create rules](#rulecreator-usage).
+
+### `devDependencies`
+
+[`@typescript-eslint/rule-tester`](../packages/Rule_Tester.mdx) is strongly recommended to be able to [test rules with our `RuleTester`](./Custom_Rules.mdx).
+
+### `peerDependencies`
+
+Include the following to enforce the version range allowed without making users' package managers install them:
+
+- `@typescript-eslint/parser` and any other parsers users are expected to be using
+- `eslint`
+- `typescript`
+
+Those are all packages consumers are expected to be using already.
+
+## `RuleCreator` Usage
+
+We recommend including at least the following three properties in your plugin's [`RuleCreator` extra rule docs types](./Custom_Rules.mdx#extra-rule-docs-types):
+
+- `description: string`: a succinct description of what the rule does
+- `recommended?: boolean`: whether the rule exists in your plugin's shared _"`recommended`"_ config
+- `requiresTypeChecking?: boolean`: whether the rule will use type information, for documentation generators such as [`eslint-doc-generator`](https://github.com/bmish/eslint-doc-generator)
+
+For example, from [`eslint-plugin-example-typed-linting`'s `utils.ts`](https://github.com/typescript-eslint/examples/blob/39a6f7bb310645c88382a25aa8bed719ccd61a0c/packages/eslint-plugin-example-typed-linting/src/utils.ts):
+
+```ts
+import { ESLintUtils } from '@typescript-eslint/utils';
+
+export interface ExamplePluginDocs {
+  description: string;
+  recommended?: boolean;
+  requiresTypeChecking?: boolean;
+}
+
+export const createRule = ESLintUtils.RuleCreator<ExamplePluginDocs>(
+  name =>
+    `https://github.com/your/eslint-plugin-example/tree/main/docs/${name}.md`,
+);
+```
+
+## Type Checking and Configs
+
+Most ESLint plugins export a _"`recommended`"_ [ESLint shared config](https://eslint.org/docs/latest/extend/shareable-configs).
+Many ESLint users assume enabling a plugin's `recommended` config is enough to enable all its relevant rules.
+
+However, at the same time, not all users want to or are able to enabled typed linting.
+If your plugin's rules heavily use type information, it might be difficult to enable those in a `recommended` config.
+
+You have roughly two options:
+
+- Have your plugin's `recommended` config require enabling type information _(recommended for new plugins)_
```

Comment:
I don't think this has anything to do with age. It's up to the plugin to decide whether the plugin needs type information to be useful at all. For example if I make a Unicorn-style plugin today I would certainly still have a separate `recommendedTypeChecked` config.